wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -24,6 +26,56 @@ import scala.util.{Failure, Success, Try}
   */
 private[wallet] trait TransactionProcessing extends WalletLogger {
   self: Wallet =>
+
+  private case object RebroadcastTransactionsRunnable extends Runnable {
+
+    override def run(): Unit = {
+      val f = for {
+        txs <- getTransactionsToBroadcast
+
+        _ = {
+          if (txs.size > 1)
+            logger.info(s"Rebroadcasting ${txs.size} transactions")
+          else if (txs.size == 1)
+            logger.info(s"Rebroadcasting ${txs.size} transaction")
+        }
+
+        _ <- nodeApi.broadcastTransactions(txs)
+      } yield ()
+
+      // Make sure broadcasting completes
+      // Wrap in try in case of spurious failure
+      Try(Await.result(f, 60.seconds))
+      ()
+    }
+  }
+
+  private[this] var rebroadcastTransactionsCancelOpt: Option[
+    ScheduledFuture[_]] = None
+
+  /** Starts the wallet's rebroadcast transaction scheduler */
+  def startRebroadcastTxsScheduler(): Unit = {
+    val future = scheduler.scheduleAtFixedRate(RebroadcastTransactionsRunnable,
+                                               30,
```

Comment:
nit: Probably make these into configuration options for the wallet. 

Something like 

`bitcoin-s.wallet.rebroadcast-frequency = 30 minutes` 

?